h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -340,7 +341,121 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
     return fr;
   }
 
-  private static RadixOrder createIndex(boolean isLeft, Frame fr, int[] cols, int[][] id_maps, int[] ascending) {
+  public static Frame sortOnly(final Frame leftFrame, final int leftCols[], int[][] id_maps, int[] ascendingL) {
+
+    createIndex(true ,leftFrame,leftCols,id_maps, ascendingL);  // sort the columns.
+    System.out.print("Making BinaryMerge RPC calls ... ");
```

Comment:
createIndex already has fixed sorting based on the sorted columns.  All we need to do next is to collect the rows into the sorted rows discovered by createIndex.

Reply:
use Log.